libnmstate/ifaces/infiniband.py

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.error import NmstateValueError
+from libnmstate.schema import InfiniBand
+from libnmstate.schema import Interface
+
+from .base_iface import BaseIface
+
+
+class InfiniBandIface(BaseIface):
+    @property
+    def parent(self):
+        return self._ib_config.get(InfiniBand.BASE_IFACE)
+
+    @property
+    def need_parent(self):
+        pkey = self.raw.get(InfiniBand.PKEY)
+        if not pkey or pkey == "0xffff":
+            return False
+        return True
+
+    @property
+    def _ib_config(self):
+        return self.raw.get(InfiniBand.CONFIG_SUBTREE, {})
+
+    @property
+    def is_virtual(self):
+        return True
+
+    def pre_edit_validation_and_cleanup(self):
+        self._validate_mandatory_properties()
+        _cannonicalize_pkey(self.raw)
+        super().pre_edit_validation_and_cleanup()
+
+    def _validate_mandatory_properties(self):
+        if self.is_up:
+            if InfiniBand.MODE not in self._ib_config:
```

Comment:
Why is mode required, if not changed?

Reply:
When not changed, the mode will copy from running state.
This only validate when user try to create new ib pkey interface.

Reply:
I am not yet convinced: https://pastebin.com/cc3LVk7M

Reply:
The scenario works for managed interfaces, so I am fine.